provisioning/baremetal_secrets_test.go

```diff
@@ -0,0 +1,87 @@
+package provisioning
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+const testNamespace = "test-namespce"
+
+func TestGenerateRandomPassword(t *testing.T) {
+	pwd, err := generateRandomPassword()
+	if err != nil {
+		t.Errorf("Unexpected error: %s", err)
+	}
+	if pwd == "" {
+		t.Errorf("Expected a valid string but got null")
+	}
+}
+
+//Testing the case where the password already exists
+func TestCreateMariadbPasswordSecret(t *testing.T) {
+	kubeClient := fakekube.NewSimpleClientset(nil...)
+	client := kubeClient.CoreV1()
+
+	// First create a mariadb password secret
+	if err := createMariadbPasswordSecret(kubeClient.CoreV1(), testNamespace); err != nil {
+		t.Fatalf("Failed to create first Mariadb password. %s ", err)
+	}
+	// Read and get Mariadb password from Secret just created.
+	oldMaridbPassword, err := client.Secrets(testNamespace).Get(context.Background(), baremetalSecretName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal("Failure getting the first Mariadb password that just got created.")
+	}
+	oldPassword, ok := oldMaridbPassword.StringData[baremetalSecretKey]
+	if !ok || oldPassword == "" {
+		t.Fatal("Failure reading first Mariadb password from Secret.")
+	}
+
+	// The pasword definitely exists. Try creating again.
+	if err := createMariadbPasswordSecret(kubeClient.CoreV1(), testNamespace); err != nil {
+		t.Fatal("Failure creating second Mariadb password.")
+	}
+	newMaridbPassword, err := client.Secrets(testNamespace).Get(context.Background(), baremetalSecretName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal("Failure getting the second Mariadb password.")
+	}
+	newPassword, ok := newMaridbPassword.StringData[baremetalSecretKey]
+	if !ok || newPassword == "" {
+		t.Fatal("Failure reading second Mariadb password from Secret.")
+	}
+	if oldPassword != newPassword {
+		t.Fatalf("Both passwords do not match.")
+	} else {
+		t.Logf("First Mariadb password is being preserved over re-creation as expected.")
+	}
+}
+
+func TestCreateMetal3PasswordSecrets(t *testing.T) {
```

Comment:
The same for this test

Reply:
Not sure how. We want to test that the top level function CreateMetal3PasswordSecrets() generates 3 passwords.

Reply:
Updated these tests after comment to remove the top level function to create all 3 passwords together.